crates/oxc_linter/src/rules/react/jsx_fragments.rs

```diff
@@ -0,0 +1,134 @@
+use oxc_ast::{
+    AstKind,
+    ast::{JSXElementName, JSXMemberExpressionObject, JSXOpeningElement},
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+use crate::{AstNode, context::LintContext, rule::Rule};
+
+fn jsx_fragments_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Shorthand form for React fragments is preferred")
+        .with_help("Use <></> instead of <React.Fragment></React.Fragment>")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct JsxFragments;
```

Comment:
I think we should support both options:

https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-fragments.md

```
...
"react/jsx-fragments": [<enabled>, <mode>]
...
```